File: Assets/GUI/ResetButton.js
#pragma strict

private static var resetCount = 0;

private var initialScale = 1.0;
private var vibe = 0.0;

function Start() {
	initialScale = transform.localScale.x;
}

function Update() {
	var param = vibe * Mathf.Sin(Time.time * 30.0);
	transform.localScale = Vector3.one * (initialScale * (1.0 + 0.5 * param));
	vibe = ExpEase.Out(vibe, 0.0, -8.0);
}

function DoReset() {
	vibe = 1.0;
	resetCount++;

	var slots = GameObject.FindGameObjectsWithTag("Slot");
	for (var i = 0; i < slots.Length; i++) {
		var slot = slots[i];
		slot.BroadcastMessage("RemoveArpies");
		slot.GetComponentInChildren.<KeyCube>().SetColor(resetCount, i);
		slot.GetComponentInChildren.<KeyAudio>().SetKey(resetCount, i);
		yield WaitForSeconds(0.03);
	}
}

File: Assets/GUI/ResetButton.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class ResetButton : MonoBehaviour
{
    private static int resetCount;
    private float initialScale;
    private float vibe;
    public virtual void Start()
    {
        this.initialScale = this.transform.localScale.x;
    }

    public virtual void Update()
    {
        float param = this.vibe * Mathf.Sin(Time.time * 30f);
        this.transform.localScale = Vector3.one * (this.initialScale * (1f + (0.5f * param)));
        this.vibe = ExpEase.Out(this.vibe, 0f, -8f);
    }

    public virtual IEnumerator DoReset()
    {
        this.vibe = 1f;
        ResetButton.resetCount++;
        GameObject[] slots = GameObject.FindGameObjectsWithTag("Slot");
        int i = 0;
        while (i < slots.Length)
        {
            GameObject slot = slots[i];
            slot.BroadcastMessage("RemoveArpies");
            slot.GetComponentInChildren<KeyCube>().SetColor(ResetButton.resetCount, i);
            slot.GetComponentInChildren<KeyAudio>().SetKey(ResetButton.resetCount, i);
            yield return new WaitForSeconds(0.03f);
            i++;
        }
    }

    public ResetButton()
    {
        this.initialScale = 1f;
    }

}